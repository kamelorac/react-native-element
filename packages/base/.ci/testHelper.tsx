import React, { JSXElementConstructor } from 'react';
import { View } from 'react-native';
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from '@testing-library/react-native';

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };
  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || 'wrapper')!;
  return { wrapper, ...renderApi };
};
