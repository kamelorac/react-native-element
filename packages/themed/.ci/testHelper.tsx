import React, { JSXElementConstructor } from 'react';
import { View } from 'react-native';
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from '@testing-library/react-native';
import {
  ThemeProvider,
  FullTheme,
  createTheme,
  CreateThemeOptions,
} from '../src/config';

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  themeProp: Partial<CreateThemeOptions> = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };
  const renderApi = render(
    <ThemeProvider theme={createTheme(themeProp)}>{children}</ThemeProvider>,
    options
  );
  const wrapper = renderApi.queryByTestId(wrapperTestID || 'wrapper')!;
  return { wrapper, ...renderApi };
};
