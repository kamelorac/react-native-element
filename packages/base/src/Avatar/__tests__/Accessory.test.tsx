import React from 'react';
import { Avatar } from '..';
import { renderWithWrapper } from '../../../.ci/testHelper';

describe('Accessory Component', () => {
  jest.useFakeTimers();
  it('should uses Icon', () => {
    const component = renderWithWrapper(<Avatar.Accessory />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('should uses Image', () => {
    const { wrapper } = renderWithWrapper(
      <Avatar.Accessory source={{ uri: 'https://i.imgur.com/0y8Ftya.jpg' }} />,
      'RNE__Image'
    );
    expect(wrapper.props.source.uri).toMatch('https://i.imgur.com/0y8Ftya.jpg');
  });
});
