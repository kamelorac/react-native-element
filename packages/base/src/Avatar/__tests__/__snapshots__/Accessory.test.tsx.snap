// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accessory Component should uses Icon 1`] = `
<View
  testID="wrapper"
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#aaa",
          "bottom": 0,
          "justifyContent": "center",
          "position": "absolute",
          "right": 0,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 0.5,
          "shadowRadius": 2,
        },
        {
          "borderRadius": 5,
          "height": 10,
          "width": 10,
        },
        undefined,
      ]
    }
  >
    <View>
      <View
        style={
          {
            "overflow": "hidden",
          }
        }
        testID="RNE__ICON__CONTAINER"
      >
        <View
          accessibilityRole="button"
          android_ripple={
            {
              "borderless": false,
              "color": "rgba(0, 0, 0, 0.3)",
              "radius": -5,
            }
          }
          disabled={false}
          testID="RNE__ICON__CONTAINER_ACTION"
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
            testID="RNE__ICON"
          >
            <Text
              allowFontScaling={false}
              style={
                [
                  {
                    "color": "#fff",
                    "fontSize": 8,
                  },
                  {
                    "backgroundColor": "transparent",
                  },
                  {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
              testID="RNE__ICON__Component"
            >
              î‰”
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
